feat(examples) : adding cache control and environmental variable changes

vnaredlawa opened the pull request:
Added cache control changes for all the products and added a change in environmental variables, reading the credentials from the System Environmental Variables.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/factset/analyticsapi-engines-java-sdk?pullRequest=66) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/factset/analyticsapi-engines-java-sdk?pullRequest=66) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/factset/analyticsapi-engines-java-sdk?pullRequest=66) it.</sub>

ksreeramoj commented:
@vnaredlawa I can see only environment variable changes, did you miss any other changes related cache control?

saigiridhar21 commented on examples/src/examples/AfiInteractiveOptimizerEngineExample.java:
```diff
@@ -59,6 +55,9 @@ public static void main(String[] args) throws InterruptedException, JsonProcessi
       
       ApiResponse<Object> response = apiInstance.postAndOptimizeWithHttpInfo(null, null, afiOptimizerParam);
+      //Comment the above line and uncomment the below lines to add cache control configuration. Results are by default cached for 12 hours; Setting max-stale=300 will fetch a cached result which is 5 minutes older.
+      //String cache_control="max-stale=300";
```
Can you please use camel casing for `cache_control`?

vnaredlawa replied:
Updated the code with camel case.

saigiridhar21 commented on examples/src/examples/AxpInteractiveOptimizerEngineExample.java:
```diff
@@ -77,6 +73,9 @@ public static void main(String[] args) throws InterruptedException, JsonProcessi
       
       ApiResponse<Object> response = apiInstance.postAndOptimizeWithHttpInfo(null, null, axpOptimizerParam);
+      //Comment the above line and uncomment the below lines to add cache control configuration. Results are by default cached for 12 hours; Setting max-stale=300 will fetch a cached result which is 5 minutes older.
+      //String cache_control="max-stale=0";
```
Can you make it `max-stale=300`?

vnaredlawa replied:
Changed the code to "max-stale=300".

saigiridhar21 commented on examples/src/examples/FiabInteractiveEngineExample.java:
```diff
@@ -47,6 +43,9 @@ public static void main(String[] args) throws InterruptedException, JsonProcessi
       calcParameters.setFisettingsdocument(FIAB_SETTINGS_DOCUMENT);
       calcParameters.setMsl(FIAB_MSL);
       ApiResponse<Void> response = apiInstance.runCalculationWithHttpInfo(calcParameters);
+      //Comment the above line and uncomment the below lines to add cache control configuration. Results are by default cached for 12 hours; Setting max-stale=300 will fetch a cached result which is 5 minutes older.
+      //String cache_control="max-stale=300";
+      //ApiResponse<Void> response = apiInstance.runCalculationWithHttpInfo(cache_control, calcParameters);
```
It think that we don't support cache_control for FIAB. Can you please crosscheck?

vnaredlawa replied:
Yes, I have removed the lines and updated the code.

saigiridhar21 commented on examples/src/examples/SPAREngineMultipleUnitExample.java:
```diff
@@ -92,6 +89,9 @@ public static void main(String[] args) throws InterruptedException, JsonProcessi
       
       ApiResponse<Object> createResponse = apiInstance.postAndCalculateWithHttpInfo(null, null, calcParameters);
+      //Comment the above line and uncomment the below lines to add cache control configuration. Results are by default cached for 12 hours; Setting max-stale=300 will fetch a cached result which is 5 minutes older.
+      //String cache_control="max-stale=0";
```
Can you make it `max-stale=300`?

vnaredlawa replied:
Changed the code to "max-stale=300".

saigiridhar21 commented on examples/src/examples/VaultEngineSingleUnitExample.java:
```diff
@@ -86,7 +83,9 @@ public static void main(String[] args) throws InterruptedException, JsonProcessi
-      
+      //Comment the above line and uncomment the below lines to add cache control configuration. Results are by default cached for 12 hours; Setting max-stale=300 will fetch a cached result which is 5 minutes older.
+      //String cache_control="max-stale=300";
+      //ApiResponse<Object> response = apiInstance.postAndCalculateWithHttpInfo(null, null, calcParameters);
```
Do you need to pass `cache-control` argument?

vnaredlawa replied:
No need to pass the argument, I have updated the code with the arguement.

vnaredlawa commented:
> @vnaredlawa I can see only environment variable changes, did you miss any other changes related cache control?

No, we have updated the changes with respect to only environmental variables and added cache_control configuration.

AkankshaAcharya commented on examples/src/examples/AfiInteractiveOptimizerEngineExample.java:
```diff
@@ -59,6 +55,9 @@ public static void main(String[] args) throws InterruptedException, JsonProcessi
       afiOptimizerParam.setData(afiItem);
       
       ApiResponse<Object> response = apiInstance.postAndOptimizeWithHttpInfo(null, null, afiOptimizerParam);
+      // Comment the above line and uncomment the below lines to add cache control configuration. Results are by default cached for 12 hours; Setting max-stale=300 will fetch a cached result which is 5 minutes older.
```
Would it be better if we make it "Setting max-stale=300 will fetch a cached result which is at max 5 minutes older"?

vnaredlawa replied:
Yes, It's working fine for 5 minutes.

saigiridhar21 replied:
@vnaredlawa Can you please make the change as suggested by @AkankshaAcharya(Include the word `max`) ?

vnaredlawa replied:
Changed the configuration lines and updated the code adding "at max" to all the products.

saigiridhar21 commented on examples/src/examples/AfiInteractiveOptimizerEngineExample.java:
```diff
@@ -30,10 +30,6 @@
 
 public class AfiInteractiveOptimizerEngineExample {
   private static FdsApiClient apiClient = null;
-  private static String BASE_PATH = "https://api.factset.com";
-  private static String USERNAME = "<username-serial>";
-  private static String PASSWORD = "<apiKey>";
```
Let's have these variables and assign the values here by reading from the environment. That way it would be clear to the client that he needs to set some values in his environment.

vnaredlawa replied:
Added the environmental variable lines for all the products.

saigiridhar21 commented on examples/src/examples/PubEngineSingleUnitExample.java:
```diff
@@ -56,8 +53,11 @@ public static void main(String[] args) throws InterruptedException, JsonProcessi
+      //Get Calculation Request Status
       ApiResponse<CalculationStatusRoot> getStatus = null;
       File result = null;
       switch (createResponse.getStatusCode()) {
```
Can you add `break` in all the case blocks like in other examples?

vnaredlawa replied:
I have added "break" for the PUB Engine single unit example, remaining all other examples are good with 'Switch-Break' statements.

saigiridhar21 reviewed (approved):
Looks good to me
